spring/spring-data/src/test/java/io/quarkus/ts/spring/data/rest/JpaRepositoryRestResourceIT.java

```diff
@@ -0,0 +1,154 @@
+package io.quarkus.ts.spring.data.rest;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.core.IsNot.not;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.http.HttpStatus;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.scenarios.QuarkusScenario;
+import io.quarkus.ts.spring.data.AbstractDbIT;
+import io.restassured.response.Response;
+
+@QuarkusScenario
+public class JpaRepositoryRestResourceIT extends AbstractDbIT {
+
+    @Test
+    void allRepositoryMethodsValidRequests() {
+        //GET - List all articles (should have all 4 articles the database has initially)
+        app.given()
+                .accept("application/json")
+                .when().get("/article-jpa")
+                .then()
+                .statusCode(HttpStatus.SC_OK)
+                .body(
+                        containsString("Aeneid"),
+                        containsString("Beach House"),
+                        containsString("Cadillac Desert"),
+                        containsString("Dagon and Other Macabre Tales"));
+
+        //GET - List all articles in hal+json format
+        String articlesList = app.given()
+                .accept("application/hal+json")
+                .when().get("/article-jpa")
+                .then()
+                .statusCode(HttpStatus.SC_OK)
+                .extract().response().jsonPath().getString("_embedded.article-jpa.name");
+        assertEquals("[Aeneid, Beach House, Cadillac Desert, Dagon and Other Macabre Tales]", articlesList);
+
+        //POST - Create a new Article
+        app.given()
+                .contentType("application/json")
+                .accept("application/json")
+                .body("{\"name\": \"Early Asimov\", \"author\": \"Isaac Asimov\"}")
+                .when().post("/article-jpa")
+                .then()
+                .statusCode(HttpStatus.SC_CREATED)
+                .body(containsString("Early Asimov"))
+                .body("id", notNullValue())
+                .extract().body().jsonPath().getString("id");
```

Comment:
What is the purpose of the last line? It looks like you get the result, but then ignore it.

Reply:
Added an assert of the ID value.